src/datasets/builder.py

```diff
@@ -813,6 +815,9 @@ def _build_single_dataset(
                         )
                     else:
                         ds.info.features = self.info.post_processed.features
+            # Rename and cast features to match task schema
+            if self.task is not None:
+                ds = ds.prepare_for_task(self.task)
```

Comment:
Maybe let's move this to the `load_dataset` function directly ? This way you don't have to add the `task` attribute to the builder itself.

Reply:
good idea!